src/bind.py

```diff
@@ -343,3 +398,37 @@ def _dns_record_relations_data_to_zones(
                 else:
                     zones[new_zone.domain] = new_zone
         return list(zones.values())
+
+    def _get_zonefile_metadata(self, zonefile_content: str) -> dict[str, str]:
+        """Get the metadata of a zonefile.
+
+        Args:
+            zonefile_content: The content of the zonefile.
+
+        Returns:
+            The hash of the corresponding zonefile.
+
+        Raises:
+            InvalidZoneFileMetadataError: when the metadata of the zonefile could not be parsed.
+            EmptyZoneFileMetadataError: when the metadata of the zonefile is empty.
+        """
+        # This assumes that the file was generated with the constants.ZONE_HEADER_TEMPLATE template
+        metadata = {}
+        try:
+            lines = zonefile_content.split("\n")
+            for line in lines:
+                # We only take the $ORIGIN line into account
+                if not line.startswith("$ORIGIN"):
+                    continue
+                logger.debug("%s", line)
+                for token in line.split(";")[1].split():
+                    k, v = token.split(":")
+                    metadata[k] = v
+                logger.debug("%s", metadata)
+                break
```

Comment:
You are just reading the first line starting with "$ORGIIN", is that intended?

Reply:
Hmmm... I thought that there could be only one `$ORIGIN` per zonefile but I don't see that anywhere. We're currently generating zonefiles with only one `$ORIGIN`, that's for sure.  
That said, I can just remove the break without any consequences on the happy path.

Reply:
I just wanted to make sure this was intended. As far as I'm concerned, you can leave it as it is

Reply:
I decided to remove the `break` because I got confirmation that multiple `$ORIGIN` are possible. I added a `DuplicateMetadataEntryError` to avoid cases where we would silently overwrite an entry and adjusted the unit tests.